src/tcgwars/logic/impl/gen4/DiamondPearl.groovy

```diff
@@ -480,16 +554,31 @@ public enum DiamondPearl implements LogicCardInfo {
           weakness F, PLUS30
           resistance P, MINUS20
           pokePower "Darkness Fan", {
-            text "Once during your turn , if Shiftry is your Active Pokémon, you may flip a coin. If heads, choose 1 Evolved Pokémon on your opponent’s Bench, remove the highest Stage Evolution card from that Pokémon, and put it back into his or her hand. This power can’t be used if Shiftry is affected by a Special Condition."
+            text "Once during your turn (before your attack), if Shiftry is your Active Pokémon, you may flip a coin. If heads, choose 1 Evolved Pokémon on your opponent’s Bench, remove the highest Stage Evolution card from that Pokémon, and put it back into his or her hand. This power can’t be used if Shiftry is affected by a Special Condition."
             actionA {
+              checkLastTurn()
+              assert self.active : "$self is not your active Pokémon."
+              assert !(self.specialConditions) : "$self is affected by a Special Condition"
+              def list = opp.bench.findAll { it.evolution }
+              assert list : "Your opponent has no evolved Pokémon."
+              powerUsed()
+              flip {
+                def pcs = list.select("Devolve one of your opponent's evolved Benched Pokémon.")
+                assert pcs
```

Comment:
Yup, agree looking at it in more detail. Will remove it from UNM's CELEBI_4 later as well...

One thing I want to check about this is how it'd interact with non-evolved evolutions. I imagine it distinguishes but it may be needed to add a separate `it.evolved` attribute for it to work properly.

Reply:
That's something I'm not totally sure of after discussing the fossil Pokemon as basics due to some of the gen 3 cards with axpendix. I think `.evolution` checks if the Pokemon is "evolved" which means there is another PCS underneath it. You may be able to directly call the correct evolution call in `PokemonCardSet.java` on a PCS directly.